ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2722,17 +2729,22 @@ private void fireInsertEvent(Table tbl, Map<String, String> partitionSpec, boole
 
 
   private static void addInsertFileInformation(List<Path> newFiles, FileSystem fileSystem,
-      InsertEventRequestData insertData) throws IOException {
+                                               InsertEventRequestData insertData, HiveConf conf) throws IOException {
     LinkedList<Path> directories = null;
+    boolean isSupportDifferentFs = conf.getBoolVar(ConfVars.HIVE_SUPPORT_MULTIPLE_FSLOCATIONS);
     for (Path p : newFiles) {
-      if (fileSystem.isDirectory(p)) {
+      FileSystem pFileSystem = fileSystem;
+      if (isSupportDifferentFs && (!(p.toUri().getScheme().equals(fileSystem.getScheme())))) {
+        pFileSystem = p.getFileSystem(conf);
+      }
+      if (pFileSystem.isDirectory(p)) {
```

Comment:
> My patch also can fix issue when "set hive.metastore.dml.events=true"

@zhixingheyi-tian sure, i know this change maybe can fix `Wrong FS` exception. But i think you need optimize this change and submit a PR to master branch. :)
Thanks.

Reply:
btw, master branch has supported the partitions on different, so no need add the new configuration `hive.support.multiple.fslocations`.  And what you need on master branch is that check&get the correct location(or filesytem) to avoid the `Wrong FS` exception when "set hive.metastore.dml.events=true"